### ISSUE TITLE
[Bug]: [curseforge-generic] modpack with name that contains spaces/special characters causes curl to fail

### ISSUE THREAD

**Issue by AestasLonewolf** [parkervcp/eggs#1540]
### Panel Version

1.5.1

### Wings Version

1.5.x

### Service

minecraft/curseforge-generic

### Modified

Yes, I modified the egg (will provide details below)

### Expected Behavior

Installing the server

### Actual Behavior

```
Executing curl -L https://edge.forgecdn.net/files/3611/633/Better Minecraft Server Pack [FABRIC] v11.5.zip -o server.zip
curl: (3) bad range in URL position 72:
https://edge.forgecdn.net/files/3611/633/Better Minecraft Server Pack [FABRIC] v11.5.zip
                                                                       ^
```

curl fails because of spaces/brackets in URL

### Steps To Reproduce

Installing a server with the curseforge-generic egg and a modpack with a name that contains special characters like 
https://www.curseforge.com/minecraft/modpacks/better-minecraft-fabric

![image](https://user-images.githubusercontent.com/86505839/150708273-48de05b6-ff34-44b1-b0d1-9ff9cef451d2.png)
 The JSON will return a URL that can't be parsed correctly with curl

### Install logs

/

**Comment by iamkubi** [comment]
You can't put spaces in a URL, you have to URL encode it just like you see you browser did automatically.  Curl does not do this automatically.

**Comment by iamkubi** [comment]
You can also just put quotes around it.

```
# curl -L "https://edge.forgecdn.net/files/3611/633/Better Minecraft Server Pack [FABRIC] v11.5.zip" -o server.zip
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100  292M  100  292M    0     0  51.7M      0  0:00:05  0:00:05 --:--:-- 51.3M
```

**Comment by LinusOP** [comment]
> You can't put spaces in a URL, you have to URL encode it just like you see you browser did automatically. Curl does not do this automatically.

The spaces are put there by the curseforge API, can confirm this happens on an unmodified version of the egg with the `Better Minecraft [FABRIC]` pack, and the egg should probably be edited to handle that curse unfortunately don't URL encode the download in their API response.

**Comment by iamkubi** [comment]
Oh, I assumed this was just because of the modified egg.  If it's also happening in unmodified then the install script can be fixed.

**Comment by AestasLonewolf** [comment]
I just tried and adding `-g` to the curl fixes this as far as I know #1541

**Comment by Software-Noob** [comment]
That's indeed caused by globbing as curl expects you to provide a range for the glob. Pretty weird for them to have such download URLs.
